pkg/sloop/ingress/dbrestore.go

```diff
@@ -17,10 +20,25 @@ func DatabaseRestore(db badgerwrap.DB, filename string) error {
 	}
 	defer file.Close()
 
-	err = db.Load(file, runtime.NumCPU())
+	zr, err := zstd.NewReader(file)
 	if err != nil {
-		return errors.Wrapf(err, "failed to restore database from file: %q", filename)
+		return errors.Wrapf(err, "failed to initialize zstd reader")
 	}
+	defer zr.Close()
+
+	err = db.Load(zr, runtime.NumCPU())
+	if errors.Is(err, zstd.ErrMagicMismatch) {
+		glog.V(2).Infof("database file is not compressed with zstd, will load without decompressing")
 
+		// We already loaded the database, advancing the file offset. To load the database again,
+		// we must reset the offset to the start.
+		if _, err := file.Seek(0, io.SeekStart); err != nil {
+			return errors.Wrapf(err, "failed to to rewind to start of database restore file: %q", filename)
+		}
+		err = db.Load(file, runtime.NumCPU())
+	}
+	if err != nil {
```

Comment:
I wanted to check if the file is compressed or not before loading the database, but could not find a way to do that using the zstd package. Attempting to load the database is the best alternative I found.